components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -427,24 +493,48 @@ func (g *rtkSerial) receiveAndWriteSerial() {
 		msg, err := scanner.NextMessage()
 		if err != nil {
 			g.ntripMu.Lock()
-			g.ntripStatus = false
+			g.isConnectedToNtrip = false
 			g.ntripMu.Unlock()
 
 			if msg == nil {
 				if g.isClosed {
 					return
 				}
+
 				g.logger.Debug("No message... reconnecting to stream...")
-				err = g.getStream(g.ntripClient.MountPoint, g.ntripClient.MaxConnectAttempts)
-				if err != nil {
-					g.err.Set(err)
-					return
+				if g.isVirtualBase {
+					g.logger.Debug("reconnecting to the Virtual Reference Station")
+					err = g.sendGGAMessage()
+
+					if err != nil && !errors.Is(err, io.EOF) {
+						g.err.Set(err)
+					}
+					if g.readerWriter == nil {
+						g.logger.Debug("Reached EOF, trying to reconnect")
```

Comment:
Logs don't obviously match code: we haven't necessarily reached EOF. Maybe the predicate in the if statement is supposed to be `errors.Is(err, io.EOF)`?